libs/indicom.c

```diff
@@ -1656,27 +1655,44 @@ double get_local_hour_angle(double sideral_time, double ra)
     return rangeHA(HA);
 }
 
-void get_alt_az_coordinates(double Ha, double Dec, double Lat, double* Alt, double *Az)
+void get_alt_az_coordinates(double Ra, double Dec, double Lat, double lst, double* Alt, double *Az)
 {
     double alt, az;
-    Ha *= M_PI / 180.0;
+    double ha = get_local_hour_angle(lst, Ra);
+    double rad = ha * M_PI / 12.0;
     Dec *= M_PI / 180.0;
     Lat *= M_PI / 180.0;
-    alt = asin(sin(Dec) * sin(Lat) + cos(Dec) * cos(Lat) * cos(Ha));
-    az = acos((sin(Dec) - sin(alt)*sin(Lat)) / (cos(alt) * cos(Lat)));
+    alt = asin(sin(Dec) * sin(Lat) + cos(Dec) * cos(Lat) * cos(rad));
+    az = asin(-sin(rad) * cos(Dec) / cos(alt));
     alt *= 180.0 / M_PI;
     az *= 180.0 / M_PI;
-    if (sin(Ha) >= 0.0)
-        az = 360 - az;
-    *Alt = alt;
-    *Az = az;
+    if(ha < -6.0 || ha > 6.0)
+        az = 180.0-az;
+    *Alt = rangeDec(alt);
+    *Az = range360(az);
+}
+
+void get_ra_dec_coordinates(double Alt, double Az, double Lat, double lst, double* Ra, double *Dec)
```

Comment:
why not use libnova functions?

Reply:
Ah.. yes.. err.. why?

Reply:
Sorry, these functions don't need external dependency, the main reason is that need arbitrary sidereal time argument. Since that's obtained by a libnova function maybe calling there libnova funcions again maybe will better